### ISSUE TITLE
@smithy/types Symbol index signature parameter breaks TS 3.8

### ISSUE THREAD

**Issue by jimblanc** [awslabs/smithy-typescript#865]
It appears that a [recent commit](https://github.com/awslabs/smithy-typescript/commit/f6cb949d1d691f8f83e8f88ff124cf121ff3c570) which was released with `@smithy/types` 2.2.0 contains a breaking change for customers using older versions of TypeScript.

We are seeing the following error with TypeScript 3.8:
```
@aws-amplify/analytics: {
@aws-amplify/analytics:   message: "/root/amplify-js/node_modules/@smithy/types/dist-types/ts3.4/extensions/checksum.d.ts (23,6): An index signature parameter type must be either 'string' or 'number'.",
@aws-amplify/analytics:   level: 'error'
@aws-amplify/analytics: }
@aws-amplify/analytics: {
@aws-amplify/analytics:   message: "/root/amplify-js/node_modules/@smithy/types/dist-types/ts3.4/extensions/checksum.d.ts (23,6): An index signature parameter type must be either 'string' or 'number'.",
@aws-amplify/analytics:   level: 'error'
@aws-amplify/analytics: }
```

Support for symbol index parameters was introduced in TS 4.4 I believe.

**Comment by permadiwibisono** [comment]
it also break in TS 4.3.5

```
node_modules/@smithy/types/dist-types/extensions/checksum.d.ts(23,6): error TS1023: An index signature parameter type must be either 'string' or 'number'.
```

**Comment by jimblanc** [comment]
Issue appears to be fixed with v2.2.1, thanks!
